### ISSUE TITLE
Upgrading Dynamitey dependency to .NET 10

### ISSUE THREAD

**Issue by AtwoodTM** [fsprojects/FSharp.Interop.Dynamic#29]
### Description

To update FSharp.Interop.Dynamic to .NET 6, one of its main dependencies must also be updated (https://github.com/ekonbenefits/dynamitey).  This issue will track that update, which will tie into another issue (to be opened) to update this project to .NET 6.  The PR link is to be added below once it is opened for Dynamitey.

**Comment by AtwoodTM** [comment]
The PR to the Dynamitely project 2+ weeks ago has not received a response.  I have forked the project and updated it to .NET 6.0/.NET 8.0 frameworks.  Therefore, I will create a new NuGet package named "dynamitey-core" for this project and push it to nuget.org.  I will then compile FSharp.Interop.Dynamic for both .NET 6.0 and .NET 8.0.  If the original project regains activity, I will merge "dynamitey-core" into the original project.  But as of now, the original project is inactive.

**Comment by dsyme** [comment]
Sounds good!

**Comment by AtwoodTM** [comment]
@dsyme — status update on this issue.

The Dynamitey continuation is ready for a 4.0.0 release. It builds for .NET Standard 2.0 and .NET 10.

Two paths from here:

1. Merge that work back into upstream Dynamitey, if the original project wants it.
2. Publish 4.0.0 at the end of September from the continuation, so this library (and anyone else still on 3.0.3) can take a current package.

FSharp.Interop.Dynamic's own modernization is on `modernize-build` ([#30](https://github.com/fsprojects/FSharp.Interop.Dynamic/issues/30)). That branch still references Dynamitey 3.0.3; switching the dependency waits on 4.0.0 being available.

The continuation lives at https://github.com/dynamitey-community/dynamitey.
